### PR TITLE
make epub file extension replacement case-insensitive

### PR DESCRIPTION
**Pull request by mojavelinux**

### CODE REVIEW

**Comment by mojavelinux** [comment]
Now if the user specifies `-o foo.MOBI`, the name of the final file with be `foo.MOBI` instead of `foo.mobi`.

**Comment by chloerei** [comment]
I think it make complex to understand what is mobi output result. Output name should base on `target` but not `epub_file`, epub_file is a temp file for mobi.

Problem come form epub converter support two format but only set `outfilesuffix` as one value `.epub`. I'm not complete read the converter call stack, but change it maybe too far for fix output filename.

I think current fix-out-file-option branch is good for fix `-o` problem.

**Comment by mojavelinux** [comment]
I see your concern. Technically, the epub_file is derived from the target, so the two files have the same rootname, but I agree that it is clearer to rebuild the path off the target directly. I have updated my pull request. I think it satisfies your concern while still maintaining compatibility with existing behavior.

(I realize that we are still lacking tests...I am testing against editions to make sure that it continues to work. Clearly, we need to get tests setup...that's a major error in my part that I skipped that step).

File: lib/asciidoctor-epub3/packager.rb [hunk 1]
```diff
@@ -459,7 +460,7 @@ def distill_epub_to_mobi epub_file, target
       require 'kindlegen' unless defined? ::Kindlegen
       kindlegen_cmd = ::Kindlegen.command
     end
-    mobi_file = ::File.basename(target).sub EpubExtensionRx, '.mobi'
+    mobi_file = ::File.basename(target =~ MobiExtensionRx ? target : %(#{::Asciidoctor::Helpers.rootname target}.mobi))
```
Comment (mojavelinux) [inline lib/asciidoctor-epub3/packager.rb]: Now we only substitute if the extension isn't `.mobi`. That's the one thing we are enforcing. The extension of the mobi file must be `.mobi` (case insensitive).

**Comment by chloerei** [comment]
I still think it's not necessary to have a guard to check target is end with /.mobi/i , `.kf8` and `.azw3` is also a valid KF8 extname. asciidoctor-pdf and kindlegen both don't prevent the use of any extname.

But I accept it that let things move on, it's not a big problem...

**Comment by mojavelinux** [comment]
Now I see why you were replacing the epub extension with mobi. It took me a while to understand why you were doing it that way, but now I totally see it. Sorry it took me so long to make the connection.

**Comment by mojavelinux** [comment]
The only change that remains is to make the epub extension regex case-insensitive. Once that's merged in, then I think we'll finally have it sorted out.

**Comment by chloerei** [comment]
Great!

**Pull request merged**